scrapy_zyte_api/providers.py

```diff
@@ -116,6 +122,20 @@ async def __call__(  # noqa: C901
             options_name = f"{kw}Options"
             if item_type not in to_provide_stripped and options_name in zyte_api_meta:
                 del zyte_api_meta[options_name]
+            elif options_name in zyte_api_meta:
+                extract_from = zyte_api_meta[options_name]["extractFrom"]
+
+        if HttpOrBrowserResponse in to_provide:
+            if extract_from == "browserHtml":
+                html_requested = True
+            elif extract_from == "httpResponseBody":
+                param_parser = _ParamParser(crawler)
+                params = param_parser.parse(request)
+                del params["url"]
+                zyte_api_meta.update(params)
```

Comment:
I didn't get this part, could you please explain the logic here?

Reply:
The line `params = param_parser.parse(request)` would return something lke:
```
{'customHttpRequestHeaders': [{'name': 'Accept',
                               'value': 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8'},
                              {'name': 'Accept-Language', 'value': 'en'}],
 'httpResponseBody': True,
 'httpResponseHeaders': True,
 'url': 'https://books.toscrape.com'}
```
which are needed for the `HttpResponse`.